docs/usage/upload_files.rst

```diff
@@ -0,0 +1,199 @@
+
+TODO: JS Uploaders
+------------------
+
+Upload files from JavaScript
+
+
+Multipart Upload
+----------------
+
+In some cases, a file may be too large for a single upload. Or we may want to speed up the upload process by uploading multiple parts in parallel.
+In these cases, we have to use multipart uploads. This requires that each part is the same size, except for the last one.
+Once all parts have been uploaded, the multipart upload completes, and the parts are merged into one single file.
+At this, point, it is no longer possible to upload new parts.
+When uploading a multipart file, if one of the chunks fails, the database is rolled back and the part is deleted.
+
+As an example, we will upload a log file. We will first split the file into 5 chunks using the linux :code:`split` command:
+
+.. code-block:: console
+
+   dd if=/dev/urandom of=my_file.txt bs=1048576 count=11
+
+   split -b6291456 my_file.txt segment_
+
+To initialize a multipart upload, we use the POST command, and specify that it is a multipart upload with the aid of the "uploads" query parameter.
+We also need to specify the total size and the part size.
+
+We create a new bucket:
+
+.. code-block:: console
+
+   $ curl -X POST http://localhost:5000/files
+
+The ID is contained in the response:
+
+.. code-block:: json
+
+    {
+       "max_file_size":null,
+       "updated":"2019-05-17T06:52:52.897378+00:00",
+       "locked":false,
+       "links":{
+          "self":"http://localhost:5000/files/c896d17b-0e7d-44b3-beba-7e43b0b1a7a4",
+          "uploads":"http://localhost:5000/files/c896d17b-0e7d-44b3-beba-7e43b0b1a7a4?uploads",
+          "versions":"http://localhost:5000/files/c896d17b-0e7d-44b3-beba-7e43b0b1a7a4?versions"
+       },
+       "created":"2019-05-17T06:52:52.897373+00:00",
+       "quota_size":null,
+       "id":"c896d17b-0e7d-44b3-beba-7e43b0b1a7a4",
+       "size":0
+    }
+
+.. code-block:: console
+
+   $ B=c896d17b-0e7d-44b3-beba-7e43b0b1a7a4
+
+   $ curl -i -X POST \
+     "http://localhost:5000/files/$B/my_file.txt?uploads&size=11534336&partSize=6291456"
+
+    {
+       "updated":"2019-05-17T07:07:22.219002+00:00",
+       "links":{
+          "self":"http://localhost:5000/files/c896d17b-0e7d-44b3-beba-7e43b0b1a7a4/my_file.txt?uploadId=a85b1cbd-4080-4c81-a95c-b4df5d1b615f",
+          "object":"http://localhost:5000/files/c896d17b-0e7d-44b3-beba-7e43b0b1a7a4/my_file.txt",
+          "bucket":"http://localhost:5000/files/c896d17b-0e7d-44b3-beba-7e43b0b1a7a4"
+       },
+       "last_part_size":5242880,
+       "created":"2019-05-17T07:07:22.218998+00:00",
+       "bucket":"c896d17b-0e7d-44b3-beba-7e43b0b1a7a4",
+       "completed":false,
+       "part_size":6291456,
+       "key":"my_file.txt",
+       "last_part_number":1,
+       "id":"a85b1cbd-4080-4c81-a95c-b4df5d1b615f",
+       "size":11534336
+    }
+
+Continue uploading parts, by using the PUT command and specifying the id of the upload:
```

Comment:
I am not sure if we should use `PUT command` or `PUT request`, since `curl` is a command and `PUT` is a type of request. What do you think?

Reply:
`PUT` is a request